src/magicbox/MagicBox.ts

```diff
@@ -235,24 +231,27 @@ export class MagicBoxInstance {
     this.inputManager.blur();
   }
 
-  public clearSuggestion() {
-    this.suggestionsManager.mergeSuggestions([], suggestions => {
-      this.updateSuggestion(suggestions);
-    });
+  public async clearSuggestion() {
     this.inputManager.setWordCompletion(null);
+    this.suggestionsManager.clearSuggestions();
+    this.onSuggestions([]);
   }
 
   private focusOnSuggestion(suggestion: Suggestion) {
     if (suggestion == null || suggestion.text == null) {
-      suggestion = this.getFirstSuggestionText();
-      this.inputManager.setResult(this.displayedResult, suggestion && suggestion.text);
+      this.inputManager.setResult(this.displayedResult, this.getFirstSuggestionText());
     } else {
       this.inputManager.setResult(this.grammar.parse(suggestion.text).clean(), suggestion.text);
     }
   }
 
-  private getFirstSuggestionText(): Suggestion {
-    return find(this.lastSuggestions, suggestion => suggestion.text != null);
+  private getFirstSuggestionWithText() {
+    return find(this.suggestionsManager.suggestions, suggestion => suggestion.text != null);
```

Comment:
Not a fan of the `!= null` here. Could maybe just be `!!suggestion.text`

Reply:
Good point, fixing that.